packages/vue/src/components/authenticator.vue

```diff
@@ -239,14 +239,24 @@ const hasTabs = computed(() => {
     actorState.value?.matches('signIn') || actorState.value?.matches('signUp')
   );
 });
+
+const hasRouteComponent = computed(() => {
+  return !(
+    state.value.matches('authenticated') ||
+    state.value.matches('idle') ||
+    state.value.matches('setup') ||
+    state.value.matches('signOut') ||
+    actorState.value?.matches('autoSignIn')
+  );
```

Comment:
On our next capacity, we should look into using `useAuthenticator` in this file so we can just use `route` instead of `state.matches`.

This will also let us to share `hasRouteComponent` between the three frameworks too. I left it as-is because the three frameworks handle routes quite differently, but this will be worth looking into after we refactor this file to use `useAuthenticator`.

Reply:
Good point. Vue never got refactored to use every `useAuthenticator` composable internally. I wasn't sure how that would affect re-rerendering everywhere. However, eventually we should do that, and test to make sure that it doesn't cause any issues.